modules/carto/src/api/request-with-parameters.ts

```diff
@@ -1,12 +1,22 @@
+import {isPureObject} from '@loaders.gl/core';
 import {CartoAPIError} from './carto-api-error';
 import {DEFAULT_HEADERS, DEFAULT_PARAMETERS, MAX_GET_LENGTH} from './common';
 import type {APIErrorContext} from './types';
 
 /**
  * Simple encode parameter
  */
-function encodeParameter(name: string, value: string | boolean | number): string {
-  return `${name}=${encodeURIComponent(value)}`;
+function encodeParameter(name: string, value: unknown): string {
+  if (Array.isArray(value)) {
+    return `${name}=${encodeURIComponent(value.join(','))}`;
+  }
```

Comment:
Potential problem here. Some array types are currently encoded with `.join(',')`, like columns. Others are encoded with `JSON.stringify`, like queryParameters. At this location in the code, we don't necessarily want to have option-specific logic...

Reply:
Resolved. Columns can safely be comma-deliminated ahead of time, for both POST and GET requests.